### ISSUE TITLE
Flame v2 breaking changes

### ISSUE THREAD

**Issue by spydon** [flame-engine/flame#1938]
# Flame v2 breaking changes

This issue doesn't mean that we are soon to release v2, it's not even on the horizon yet.
In here we'll track breaking changes that we would want to do for Flame v2 so that they are not forgotten.

- [ ] Rename `HasGameRef` to `HasGameReference`.
- [ ] Convert the use of `ShapeComponent` in the collision detection to use pure shapes instead.
- [ ] Remove the `Camera` from `FlameGame` in favor on the `CameraComponent` (and remove `BareFlameGame`).

Once we are ready to start working at v2 we'll convert this task list into issues.

**Comment by MuZiLee** [comment]
Looking forward to the v2

**Comment by DevinXiaoBai** [comment]
Can you consider building wechat small game running environment

**Comment by spydon** [comment]
> Can you consider building wechat small game running environment

This is unrelated to V2 breaking changes, please open a separate issue for this.

**Comment by DevinXiaoBai** [comment]
ok，Thank you for your reply




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "flame-engine/flame"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2023年2月9日(星期四) 下午3:06
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [flame-engine/flame] Flame v2 breaking changes (Issue #1938)





  
Can you consider building wechat small game running environment
  
This is unrelated to V2 breaking changes, please open a separate issue for this.

**Comment by guyluz11** [comment]
> Once we are ready to start working at v2 we'll convert this task list into issues.

What holding us back?, there is only one task that got converted into an issue

**Comment by NashIlli** [comment]
Hello, @spydon! Is Version 2 included in the route of the following months? Thaaanks! :)

**Comment by spydon** [comment]
> Hello, @spydon! Is Version 2 included in the route of the following months? Thaaanks! :)

Hi, unfortunately not, we have not even decided on paper how the big architectural changes will look.

**Comment by marzoukali** [comment]
Any updates on when to expect version 2 or do you need any help getting it back on track? We can share with the community the challenges and build separate team dedicated for it. I suggest if this is still in plan, we will need to have it's effort as separate from the current.

**Comment by ufrshubham** [comment]
> we will need to have it's effort as separate from the current

I am not sure of it will be a good idea to have separate efforts for this. Although v2 will have breaking changes, I'd still want the transition to be as smooth as possible. Maintaining two separate versions might drastically change the APIs causing users to re-write a significant portion of their game.

**Comment by spydon** [comment]
@marzoukali we have some bigger architectural changes that we need to decide and plan internally before we start this project.

**Comment by NashIlli** [comment]
> [@marzoukali](https://github.com/marzoukali) we have some bigger architectural changes that we need to decide and plan internally before we start this project.

@spydon, regarding the roadmap for Flame v2, are there any plans to implement native integration with the Rive Renderer, or will Flame continue to operate as a layer on top of Flutter's standard graphics pipeline (Skia/Impeller)?

I'm curious if a more direct bridge is being considered to leverage Rive's new hardware-accelerated capabilities directly within the Flame game loop. What are your thoughts on the long-term architectural direction for this?

**Comment by spydon** [comment]
@NashIlli no, we don't plan to use the rive renderer directly in flame. Possibly we'll add support for it in flame_rive though.

**Comment by NashIlli** [comment]
@spydon thanks for the clarification. To confirm, does this mean support will remain limited to Flutter’s renderer, or do you envision flame_rive eventually being able to leverage the full feature set and performance optimizations of the Rive renderer indirectly?

**Comment by spydon** [comment]
@NashIlli we have spoken about making a generic canvas interface so that the user can easily swap out rendering, but it's nothing that is planned for the near future.

**Comment by naychrist** [comment]
+1 for the ability to swap out rendering for rive through a generic canvas interface :)

loving the recent commits I've seen re Box2d v3!

**Comment by spydon** [comment]
> loving the recent commits I've seen re Box2d v3!

@naychrist flame_forge2d 0.20.0 was released yesterday and that uses forge2d 0.15.1 which is using Box2D v3!
It's wild how much more efficient it is.
